core/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -560,7 +569,8 @@ private void updateShard(DiscoveryNodes nodes, ShardRouting shardRouting, Shard
                         allocationIdsForShardsOnNodesThatUnderstandSeqNos(indexShardRoutingTable.activeShards(), nodes);
                 final Set<String> initializingIds =
                         allocationIdsForShardsOnNodesThatUnderstandSeqNos(indexShardRoutingTable.getAllInitializingShards(), nodes);
-                shard.updatePrimaryTerm(clusterState.metaData().index(shard.shardId().getIndex()).primaryTerm(shard.shardId().id()));
+                shard.updatePrimaryTerm(clusterState.metaData().index(shard.shardId().getIndex()).primaryTerm(shard.shardId().id()),
+                    null);
```

Comment:
ouch. We might cover this with an integration test that checks that a resync effectively happened (similar test as `testResyncAfterPrimaryPromotion`, but this time in a real cluster, not just ReplicationGroup). But that is also a form of duplication :/